exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/_mapping.py

```diff
@@ -0,0 +1,135 @@
+# https://github.com/open-telemetry/opentelemetry-specification/blob/v1.33.0/specification/compatibility/prometheus_and_openmetrics.md#metric-metadata-1
+_UNIT_MAPPINGS = {
+    # Time
+    "d": "days",
+    "h": "hours",
+    "min": "minutes",
+    "s": "seconds",
+    "ms": "milliseconds",
+    "us": "microseconds",
+    "ns": "nanoseconds",
+    # Bytes
+    "By": "bytes",
+    "KiBy": "kibibytes",
+    "MiBy": "mebibytes",
+    "GiBy": "gibibytes",
+    "TiBy": "tibibytes",
+    "KBy": "kilobytes",
+    "MBy": "megabytes",
+    "GBy": "gigabytes",
+    "TBy": "terabytes",
+    # SI
+    "m": "meters",
+    "V": "volts",
+    "A": "amperes",
+    "J": "joules",
+    "W": "watts",
+    "g": "grams",
+    # Misc
+    "Cel": "celsius",
+    "Hz": "hertz",
+    # TODO: this conflicts with the spec but I think it is correct. Need to open a spec issue
+    "1": "",
+    "%": "percent",
+}
+# Similar to _UNIT_MAPPINGS, but for "per" unit denominator.
+# Example: s => per second (singular)
+# Copied from https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/80317ce83ed87a2dff0c316bb939afbfaa823d5e/pkg/translator/prometheus/normalize_name.go#L58
+_PER_UNIT_MAPPINGS = {
+    "s": "second",
+    "m": "minute",
+    "h": "hour",
+    "d": "day",
+    "w": "week",
+    "mo": "month",
+    "y": "year",
+}
+
+
+def sanitize_full_name(name: str) -> str:
+    """sanitize the given metric name according to Prometheus rule, including sanitizing
+    leading digits
+
+    https://github.com/open-telemetry/opentelemetry-specification/blob/v1.33.0/specification/compatibility/prometheus_and_openmetrics.md#metric-metadata-1
+    """
+    # Leading number special case
+    if name and name[0].isdigit():
+        name = "_" + name[1:]
+    return _sanitize_name(name)
+
+
+def _sanitize_name(name: str) -> str:
+    """sanitize the given metric name according to Prometheus rule, but does not handle
+    sanitizing a leading digit."""
+    return _SANITIZE_NAME_RE.sub("_", name)
+
+
+def sanitize_attribute(key: str) -> str:
+    """sanitize the given metric attribute key according to Prometheus rule.
+
+    https://github.com/open-telemetry/opentelemetry-specification/blob/v1.33.0/specification/compatibility/prometheus_and_openmetrics.md#metric-attributes
+    """
+    # Leading number special case
+    if key and key[0].isdigit():
+        key = "_" + key[1:]
+    return _SANITIZE_ATTRIBUTE_KEY_RE.sub("_", key)
+
+
+def map_unit(unit: str) -> str:
+    """Maps unit to common prometheus metric names if available and sanitizes any invalid
+    characters
+
+    See:
+    - https://github.com/open-telemetry/opentelemetry-specification/blob/v1.33.0/specification/compatibility/prometheus_and_openmetrics.md#metric-metadata-1
+    - https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.101.0/pkg/translator/prometheus/normalize_name.go#L108
+    """
+    # remove curly brace UCUM annotations
+    unit = _UCUM_ANNOTATION_CURLY.sub("", unit)
```

Comment:
Might be a dumb question but do we cover the use case of `{unit with space}` here? Doesn't this turn the unit into `unitwithspace` since `_UCUM_ANNOTATION_CURLY` doesn't include blank spaces?

Reply:
That's a good catch. Technically that would be an invalid UCUM unit but OTel doesn't care.

Let me make the regex more permissive

Reply:
I fixed it and added test cases.